File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
def solution(n):
    answer = []
    
    def hanoi(src, tgt, inter, n): # 인자 순서 넣어주는 게 좀 헷갈렸음.
        if n == 1:
            answer.append([src, tgt])
        else:
            hanoi(src,inter,tgt,n-1)
            hanoi(src,tgt,inter,1)
            hanoi(inter,tgt,src,n-1)
            
    hanoi(1,3,2,n)
    
    return answer

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.js
function solution(n) {
  let answer = [];
  
  const hanoi = (n, start, mid, end) => {
    if (n === 1) answer.push([start,end])
    else {
      hanoi(n-1, start, end, mid)
      answer.push([start,end])
      hanoi(n-1, mid, start, end)
    }
  }
  
  hanoi(n, 1, 2, 3)
  return answer
}